### PR TITLE
Fix panics in packet queries 

### PR DESCRIPTION
**Pull request by adizere**
Closes: #521

## Description

______

For contributor use:

- [x] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: relayer-cli/src/commands/query/packet.rs [hunk 1]
```diff
@@ -224,9 +255,29 @@ impl Runnable for QueryUnreceivedPacketsCmd {
...
+                    opts.channel_id,
+                    src_chain.config().id,
+                    e
+                ))
+                .exit();
+            }
+        };
+
+        info!(
+            "Fetched from source chain {} the following channel {:#?}",
```
Comment (ancazamfir) [inline relayer-cli/src/commands/query/packet.rs:277]: This distracts a bit (make it debug?) and the pretty format doesn't show well. 
```suggestion
        debug!(
            "Fetched from source chain {} the following channel {:?}",
```

File: relayer-cli/src/commands/query/packet.rs [hunk 2]
```diff
@@ -224,9 +255,29 @@ impl Runnable for QueryUnreceivedPacketsCmd {
...
+        let channel_res = src_chain
             .query_channel(&opts.port_id, &opts.channel_id, Height::zero())
-            .unwrap();
+            .map_err(|e| Kind::Query.context(e));
+
+        let channel = match channel_res {
+            Ok(c) => c,
+            Err(e) => {
+                return Output::error(format!(
+                    "failed to find the target channel ({}/{}) on src chain ({}) with error: {}",
```
Comment (ancazamfir) [inline relayer-cli/src/commands/query/packet.rs:266]:
```suggestion
                    "failed to find channel ({}/{}) on chain ({}) with error: {}",
```

File: relayer-cli/src/commands/query/packet.rs [hunk 3]
```diff
@@ -356,7 +434,12 @@ impl QueryPacketAcknowledgmentCmd {
     ) -> Result<(ChainConfig, QueryPacketOptions, Sequence), String> {
         let dest_chain_config = config
             .find_chain(&self.chain_id.parse().unwrap())
-            .ok_or_else(|| "missing destination chain configuration".to_string())?;
+            .ok_or_else(|| {
+                format!(
+                    "missing configuration for the given chain ({}) ",
```
Comment (ancazamfir) [inline relayer-cli/src/commands/query/packet.rs:439]:
```suggestion
                    "missing configuration for chain ({}) ",
```

File: relayer-cli/src/commands/query/packet.rs [hunk 4]
```diff
@@ -429,11 +525,21 @@ impl QueryUnreceivedAcknowledgementCmd {
     ) -> Result<(ChainConfig, ChainConfig, QueryPacketOptions), String> {
         let src_chain_config = config
             .find_chain(&self.src_chain_id.parse().unwrap())
-            .ok_or_else(|| "missing destination chain configuration".to_string())?;
+            .ok_or_else(|| {
+                format!(
+                    "missing configuration for the src chain ({}) ",
```
Comment (ancazamfir) [inline relayer-cli/src/commands/query/packet.rs:530]:
```suggestion
                    "missing configuration for chain ({}) ",
```

File: relayer-cli/src/commands/query/packet.rs [hunk 5]
```diff
@@ -429,11 +525,21 @@ impl QueryUnreceivedAcknowledgementCmd {
...
+                    self.src_chain_id
+                )
+            })?;
 
         let dst_chain_config = config
             .find_chain(&self.dst_chain_id.parse().unwrap())
-            .ok_or_else(|| "missing destination chain configuration".to_string())?;
+            .ok_or_else(|| {
+                format!(
+                    "missing configuration for the dst chain ({}) ",
```
Comment (ancazamfir) [inline relayer-cli/src/commands/query/packet.rs:539]:
```suggestion
                    "missing configuration for chain ({}) ",
```

File: relayer-cli/src/commands/query/packet.rs [hunk 6]
```diff
@@ -462,9 +568,29 @@ impl Runnable for QueryUnreceivedAcknowledgementCmd {
...
+                    opts.channel_id,
+                    src_chain.config().id,
+                    e
+                ))
+                .exit();
+            }
+        };
+
+        info!(
+            "Fetched from src chain {} the following channel {:#?}",
```
Comment (ancazamfir) [inline relayer-cli/src/commands/query/packet.rs:590]:
```suggestion
        debug!(
            "Fetched from src chain {} the following channel {:?}",
```

**Review by ancazamfir — COMMENTED** [review]
Looks good! I have a few minor comments on the error strings and ouput, trying to give just the right amount of information, no more/less than needed. Doesn't apply to this PR specifically, but in general I still find the output of our commands hard to read, sometimes it has information that is not helpful, other times not enough and I have a significant contribution here. We should sit down and discuss this for V1.

**Review by ancazamfir — APPROVED** [review]

File: relayer-cli/src/commands/query/packet.rs [hunk 7]
```diff
@@ -40,7 +41,12 @@ impl QueryPacketCommitmentsCmd {
     ) -> Result<(ChainConfig, QueryPacketOptions), String> {
         let dest_chain_config = config
             .find_chain(&self.chain_id.parse().unwrap())
-            .ok_or_else(|| "missing destination chain configuration".to_string())?;
+            .ok_or_else(|| {
+                format!(
+                    "missing configuration for the given chain ({}) ",
```
[the remark this answers is not in the document]
Comment (adizere) [reply relayer-cli/src/commands/query/packet.rs:46]: In parallel with your comments here, I had similar thoughts with respect to these error messages and fixed (hopefully all of them).

The approach I used is 

https://github.com/informalsystems/ibc-rs/blob/ffefc34c148bd9e56874ffa9250242a438012716/relayer-cli/src/commands/query/channel.rs#L56

See https://github.com/informalsystems/ibc-rs/pull/559

[the remark this answers is not in the document]
Comment (adizere) [reply relayer-cli/src/commands/query/packet.rs]: Also, for this file specifically `relayer-cli/src/commands/query/packet.rs`, I agree with your suggestions and applied them bac733d.

**Pull request merged**